Create code flows with opensource.md (#31)
flossway/flossway: source/_posts/code flows with opensource.md

> 

Reviewer: Please add metadata as other posts do.

Author: ---
title:  代码穿流-迷失的开源乌托邦之三
date: 2022-02-25
author: 狄安
categories: 狄安
tags:
    - 开源
    - 开源协同
---

> 

Author: ---
title: 大教堂过时了吗
date: 2022-02-18
author: 狄安
categories: 狄安
tags:
    - 开源
    - 开源协同
---

玩开源的程序员，大多了解 Linus 曾经说过的那句霸气无比的话，“Talk is cheap. Show me the code”。

> 

Reviewer: ```suggestion

<!-- more -->

```

You may add a summary mark. See also other posts also.

> # 代码胜于雄辩

Reviewer: ```suggestion
## 代码胜于雄辩
```

Heading 1 is for title and reserved for content.

> 这是Linus 于 2000 年 8 月 25 日在社区邮件列表上讨论关于 Linux 内核线程优化问题时，其中一个人提出了个他认为非常高效的方案，而 Linus 认为这个方案不够好，于是他在回信中就写下了这句在今天足以堪称经典的话，这也令很多程序员感觉非常惬意。

Reviewer: ```suggestion
这是 Linus 于 2000 年 8 月 25 日在社区邮件列表上讨论关于 Linux 内核线程优化问题时，其中一个人提出了个他认为非常高效的方案，而 Linus 认为这个方案不够好，于是他在回信中就写下了这句在今天足以堪称经典的话，这也令很多程序员感觉非常惬意。
```

> 

Author: <!-- more -->

从此，每当程序员们产生各种言语纷争而又一时无果时，往往便以这句话来结束唾沫横飞的争论。一方面，我们看到的是程序员文化中“与其你口头zhuangbility，不如代码 Niuabiity”的内涵；另一方面，我们也可这样理解：代码胜过一切含糊不清的争论。因为平常的语言沟通，即使一个明确的想法，即使用同样的字词，却因为语境或者语法造句等因素，让自己原本想表达的构想无法被对方精准理解和捕捉，从而产生各种歧义和误解。而写出来的程序因为代码规则的标准性和编译后的可运行性产生了所见即所得的效果，让代码就不言自证了其原先想法的清晰性，正确性或者优越性。当一切用可见的代码及运行结果来说话，争论自然就偃旗息鼓。

Facebook 创始人扎克伯格也曾在自己公司上市的前一天发表的上市演讲中说过了类似的话，他当是这样说的：“Code wins argument”，即“代码胜于雄辩。” 其言下之意是想说两层意思：1.代码是一种比平常语言更能精确表达思想的工具。2.代码运行出来的结果可以见证想法的恰当与否。

于此，我们必须接受和承认这样一个事实：在沟通误解和语言歧义的困境面前，程序代码比日常语言有着更加精准表达的优势。程序员拥有着比普通人更好的沟通载体。这个载体就叫程序代码。代码首先是一种能清晰和精准表达思想的语言载体，其次是可以通过编译后运行来直接验证效果的沟通工具。如果说数学是最精准而抽象的语言，那么程序代码可以说是精准而同时有事半具象半抽象的语言。

> 

Reviewer: ```suggestion
```

# 什么是代码穿流？

> 

Reviewer: ```suggestion
```

开源因为继承了由理查德斯托曼所开启的自由软件运动以来所倡导的代码开放共享精神，使得程序员们可以在拥有了代码语言进行精准表达自己构想的基础上，更可以通过这些开放且透明可见的源代码承载着他们的思想，在不同的程序员间或者开源社区的不同成员之间来回穿梭，以此充分交换和反馈各自自己的想法。我把它这个过程定义为代码穿流。而这样的代码穿流正是因为代码语言的精确性，成为了开源社区不同社区成员间的低歧义和几乎零误解的沟通载体，创造了开源社区独特的高效且有效的沟通基础。

一般我们认为，开源共同体的创新基础在于分布式的社会化协作。实际上，社会化协作是代码穿流后的一种结果表现。而开源的代码语言作为沟通基础，辅之以如 GitHub 一类的开源协作平台工具，直接效果当然是提高了开发效率，但究其本质其实应该是代码穿流后形成了开源共同体间的创新价值。

为什么这样说呢？

> # 知识的显性化和去粘性

Reviewer: ```suggestion
## 知识的显性化和去粘性
```

Heading 1 is for title and reserved for content.

在知识创新的领域里，实现价值创新的一个重要基础是知识的可传递。知识作为一个特殊而无形的物品，其可传递的核心是解决两个问题

> 1）知识的显性化
> 2）知识的去粘性

Reviewer: ```suggestion

1. 知识的显性化
2. 知识的去粘性
```

进而实现知识的无损耗传递。程序代码则作为人类知识领域的一部分，当然也存在同样的显性化和去粘性问题。

人类的显性知识是可以通过文字、数据、代码、图像等方式进行传递和学习的；而隐形知识则属于人们在长期的学习工作实践中获得并积累下来，与个体的经历及体验有关，或者只能从结果推测，且大多只能意会却难以用语言明示或学习的知识。
对于显性知识下的隐形知识，在翻译美国作家菲茨•吉拉德的名作《了不起的盖茨比》时，作为译者的日本著名作家春上村树在对于如何翻译好开头和结尾、该如何表达原作者想在语言中表达的情绪时，曾经如此纠结过：

> 读上几遍还是只能感慨文字的美，每一个词都有丰富的含义和实质，既有深刻的暗示性，犹如乙醚般清淡，想要捕捉时它却从指尖溜走。 从这段评论中，我们可以看出在日常语言下，即使对于一个专业的作家而言，知识显性化的提取也是非常高难度。当然，菲茨•吉拉德用显性语言来故意蕴藏他想表达的隐含之美是另外一层艺术创作境界，而对于可以我们得出的一个结论是：日常语言对于知识的显性化传递是无法做到足够清晰的，同时，那些受高度语境限制且不确定的语言则更会成为粘性知识。所谓粘性知识是指必须通过近距离的观察和沟通甚至面对面的jiaotan通才能了解获得的知识。

Reviewer: ```suggestion

> 读上几遍还是只能感慨文字的美，每一个词都有丰富的含义和实质，既有深刻的暗示性，犹如乙醚般清淡，想要捕捉时它却从指尖溜走。 从这段评论中，我们可以看出在日常语言下，即使对于一个专业的作家而言，知识显性化的提取也是非常高难度。当然，菲茨•吉拉德用显性语言来故意蕴藏他想表达的隐含之美是另外一层艺术创作境界，而对于可以我们得出的一个结论是：日常语言对于知识的显性化传递是无法做到足够清晰的，同时，那些受高度语境限制且不确定的语言则更会成为粘性知识。所谓粘性知识是指必须通过近距离的观察和沟通甚至面对面的jiaotan通才能了解获得的知识。
```

>  大家应该知道，有个一个古老的游戏叫“拷贝不走样”。百度百科中是这样描述的：规则是先给第一个人看他要表演的话语或素材，第二个表演给第三个看，依次下传。最后一人要把他理解的意思讲出来，猜中则胜利。该游戏起源于曾经在上海电视台热播的综艺节目《智力大冲浪》。节目最早的版本其实是，人们隔着厚厚的移动墙向下一个选手传与该词相关的动作，然后最后一个人说出第一个人做的姿势的含义，往往笑倒一片观众，拷贝不走样的游戏性在于“拷贝走样”。在活动中，观众可以亲眼目睹动作渐渐走样，这是因为，清楚知道词语的 B 不一定能表达相同信息给后面的C，动作的模仿中关键的“细节”容易遗漏，或是被C、D、E等人再创造而传承。

Reviewer: ```suggestion
 大家应该知道，有个一个古老的游戏叫“拷贝不走样”。百度百科中是这样描述的：规则是先给第一个人看他要表演的话语或素材，第二个表演给第三个看，依次下传。最后一人要把他理解的意思讲出来，猜中则胜利。该游戏起源于曾经在上海电视台热播的综艺节目《智力大冲浪》。节目最早的版本其实是，人们隔着厚厚的移动墙向下一个选手传与该词相关的动作，然后最后一个人说出第一个人做的姿势的含义，往往笑倒一片观众，拷贝不走样的游戏性在于“拷贝走样”。在活动中，观众可以亲眼目睹动作渐渐走样，这是因为，清楚知道词语的 B 不一定能表达相同信息给后面的 C，动作的模仿中关键的“细节”容易遗漏，或是被 C、D、E 等人再创造而传承。
```

> 实际上，这个游戏的隐含意义就是粘性知识在信息的传递过程中，由于传递者自身及语言，动作姿势的模糊性及多义性，信息会被迅速衰减。从这个游戏的结果，我们大致可以得出一个粗浅的结论：粘性知识的传播是距离的衰减函数。
> 编译成二进制且不提供源代码的程序，虽然通过运行可以看到显性结果或者使用，但事实上因为代码的不可见就是一种典型的隐性且粘性的知识。传统的软件提供商在提供可运行的软件时会附带软件使用手册或者技术文档来局部显性化，也会通过一些面对面的交流沟通（这些往往以培训，现场指导等形式出现）来局部去粘性。但最直接和最透彻的显性化无疑就是提供源代码。当初自由软件运动和随后兴起的开源运动的初心，就是为了让代码知识显性化的无损穿流。

Reviewer: ```suggestion

实际上，这个游戏的隐含意义就是粘性知识在信息的传递过程中，由于传递者自身及语言，动作姿势的模糊性及多义性，信息会被迅速衰减。从这个游戏的结果，我们大致可以得出一个粗浅的结论：粘性知识的传播是距离的衰减函数。

编译成二进制且不提供源代码的程序，虽然通过运行可以看到显性结果或者使用，但事实上因为代码的不可见就是一种典型的隐性且粘性的知识。传统的软件提供商在提供可运行的软件时会附带软件使用手册或者技术文档来局部显性化，也会通过一些面对面的交流沟通（这些往往以培训，现场指导等形式出现）来局部去粘性。但最直接和最透彻的显性化无疑就是提供源代码。当初自由软件运动和随后兴起的开源运动的初心，就是为了让代码知识显性化的无损穿流。
```

>   
> 

Reviewer: ```suggestion
  
```

所以开源软件实践中，程序员通过代码编写将自己的想法精确表达出来，并且通过代码开源行动开放共享出来，就是一次非常典型的知识显性化过程，随后通过社区开展的各类活动如论坛,meetup,会议等线上线下活动进行知识的去粘性化，进而让显性化的代码知识在开源社区的不同成员间无损穿流达到分享传递和协同，让不同程序员的匠性得以聚合，最终达到社会化协同创新。

> # 我们的结论 

Reviewer: ```suggestion
## 我们的结论 
```

Heading 1 is for title and reserved for content.

所以，软件代码通过开源让软件程序的隐形知识显性化加强，显性化的知识进而通过开源社区的各类活动以最大限度去粘性化，形成代码穿流。这，应是开源集市模式卓越性的一个本质所在。这也就是代码穿流力。由此，我们可以得出第一个结论：

> ## 结论1 ：开源的软件代码是软件类知识传递损耗系数最小的传递载体。
> 软件代码的开源是软件知识显性化的最优方式，而积极的开源社区活动则是软件知识去粘性化的最佳手段。

Reviewer: ```suggestion
### 结论 1：开源的软件代码是软件类知识传递损耗系数最小的传递载体

软件代码的开源是软件知识显性化的最优方式，而积极的开源社区活动则是软件知识去粘性化的最佳手段。
```

> 前不久，Apache基金会主席David Nelly在美国参议院的听证会上所言中看到，开源已经成为数字经济的重要基础设施。那么程序代码必然是开源创新协同的重要载体。Linux基金会的执行董事Jim Zemlin曾经在2013年的一次TEDx演讲中提到这样一些数据：荷马史诗《伊利亚德》15,000行，《战争与和平》45万字，在Linux项目中，每一天的每一小时至少有7个变化在 Linux项目里发生，仅2012年就有近一百万行的代码被提交，这又是由分布在全球的407个公司、几千个程序员一起合作产生的结果。到今天，Linux毫无疑义的已经成为在过去30年以来最成功的全球化开源协作项目。正是这些Linux里开源的代码穿越了时间，跨越了地理空间，更涵盖了不同种族的人群，进而造就了人类软件史上的一次伟大奇迹，这几乎和代码穿流的力量密不可分。由此，我们可以得到了代码穿流的另一个结论：

Reviewer: ```suggestion
前不久，Apache 基金会主席 David Nelly 在美国参议院的听证会上所言中看到，开源已经成为数字经济的重要基础设施。那么程序代码必然是开源创新协同的重要载体。Linux 基金会的执行董事 Jim Zemlin 曾经在 2013 年的一次 TEDx 演讲中提到这样一些数据：荷马史诗《伊利亚德》15,000 行，《战争与和平》45 万字，在 Linux 项目中，每一天的每一小时至少有 7 个变化在 Linux 项目里发生，仅 2012 年就有近一百万行的代码被提交，这又是由分布在全球的 407 个公司、几千个程序员一起合作产生的结果。到今天，Linux 毫无疑义地已经成为在过去 30 年以来最成功的全球化开源协作项目。正是这些 Linux 里开源的代码穿越了时间，跨越了地理空间，更涵盖了不同种族的人群，进而造就了人类软件史上的一次伟大奇迹，这几乎和代码穿流的力量密不可分。由此，我们可以得到了代码穿流的另一个结论：
```

> 
> 
> ## 结论2.开源的代码在时间，空间和不同人群之间穿流的程度决定了这个开源社区的价值。 

Reviewer: ```suggestion

### 结论 2：开源的代码在时间，空间和不同人群之间穿流的程度决定了这个开源社区的价值
```

我们也可以这样认为，在开源项目原始创意价值确定的前提下，开源社区的价值是代码穿流的时间，空间和人群的函数。即如果代码所穿越的群体越广泛、时间和空间，那么价值可能就越大。而它的逆否命题则可以这样描述，如果一个开源社区无法在跨时空和跨群体里穿越，那么它的创新价值有限。

> 
> 

Reviewer: ```suggestion

```

> 代码穿流后，软件代码知识就不再是时间和空间的衰减函数，反而可能成为时间和空间的递增函数。进一步而言，开源社区也未必会因为多样化成岩和多元化角色间的差异衰减社区价值，它们可能表现为随着社区成员多样化和多元化，并随着成员间的争论增多而更显得有价值。那时，如果开源社区的代码如果想更快更新迭代，也未必需要社区成员很擅长懂得如何协作，也未必需要每个社区成员成为一个不敢争论的“老好人”，社区可能更多需要的是能够经常进行相互批评并能捍卫自己的想法的成员，当然这里有一个必要的前提就是对于所有的争论，最后能够以 show me the code 来终结，而show me the code 也足以让争论分出胜负。也许我的这个想法有些乌托邦，但事实可能也就如此。

Reviewer: ```suggestion
代码穿流后，软件代码知识就不再是时间和空间的衰减函数，反而可能成为时间和空间的递增函数。进一步而言，开源社区也未必会因为多样化成岩和多元化角色间的差异衰减社区价值，它们可能表现为随着社区成员多样化和多元化，并随着成员间的争论增多而更显得有价值。那时，如果开源社区的代码如果想更快更新迭代，也未必需要社区成员很擅长懂得如何协作，也未必需要每个社区成员成为一个不敢争论的“老好人”，社区可能更多需要的是能够经常进行相互批评并能捍卫自己的想法的成员，当然这里有一个必要的前提就是对于所有的争论，最后能够以 show me the code 来终结，而 show me the code 也足以让争论分出胜负。也许我的这个想法有些乌托邦，但事实可能也就如此。
```

> 而关于激烈辩论有利于创新这一点，在2003年伯克利大学曾经做过一次称作"一个想法是如何被创造出来的？的实验。当时，一些实验者被分成两组，其中一个组采用“传统的头脑风暴法”，即该组成员不相互批评、而鼓励每个被提出来的想法，认为没有一个想法是坏的想法。另外一个组则采用"激烈辩论法"，即小组成员相互批评各自想法，鼓励以积极的辩论去捍卫自己的想法。出乎我们的意料，这个实验最后的结果竟然是激烈辩论法的那一组获胜，相比头脑风暴组而言，他们反而产生了高出头脑风暴组一个数量级的好想法。由此，我们又得出第三个结论

Reviewer: ```suggestion
而关于激烈辩论有利于创新这一点，在 2003 年伯克利大学曾经做过一次称作“一个想法是如何被创造出来的？”的实验。当时，一些实验者被分成两组，其中一个组采用“传统的头脑风暴法”，即该组成员不相互批评、而鼓励每个被提出来的想法，认为没有一个想法是坏的想法。另外一个组则采用“激烈辩论法”，即小组成员相互批评各自想法，鼓励以积极的辩论去捍卫自己的想法。出乎我们的意料，这个实验最后的结果竟然是激烈辩论法的那一组获胜，相比头脑风暴组而言，他们反而产生了高出头脑风暴组一个数量级的好想法。由此，我们又得出第三个结论：
```

> ## 结论 3.  开源社区内的激烈争论将有利于开源软件的创新。

Reviewer: ```suggestion
### 结论 3：开源社区内的激烈争论将有利于开源软件的创新
```

> 当然，这基于代码语言的精准表达性和代码编译运行后的可校验性的前提。
> 
> Jim Zemlin在那年的TEDx演讲中还如此动情地讲到：>All of us are smarter than any one of us.  there is a whole generation of code poets out there, working furiously. Poets who love what they do, they may not get along, but are creating ... These indivuals have created the coal and steel of the informaiton age. ...... The future is a world in which can enrich your self, while at the same time enriching others.

Reviewer: ```suggestion

当然，这基于代码语言的精准表达性和代码编译运行后的可校验性的前提。

Jim Zemlin 在那年的 TEDx 演讲中还如此动情地讲到：

> All of us are smarter than any one of us.  there is a whole generation of code poets out there, working furiously. Poets who love what they do, they may not get along, but are creating ... These individuals have created the coal and steel of the information age ... The future is a world in which can enrich your self, while at the same time enriching others.
```

这就是代码穿流的威力。

> 也恰如关于乔布斯生前往事的纪录片《失落的访谈》中曾记录了乔布斯在1995年接受的采访中所提到的一个观点，程序语言在未来应当像其他我们日常所用来沟通的语言一样，成为我们每个人必备的交流工具。虽然这对一般普通人要求过高，但无疑他看到了代码语言作为思想的传达载体具有比数学更具象，比语言更精准的表达。

Reviewer: ```suggestion
也恰如关于乔布斯生前往事的纪录片《失落的访谈》中曾记录了乔布斯在 1995 年接受的采访中所提到的一个观点，程序语言在未来应当像其他我们日常所用来沟通的语言一样，成为我们每个人必备的交流工具。虽然这对一般普通人要求过高，但无疑他看到了代码语言作为思想的传达载体具有比数学更具象，比语言更精准的表达。
```

> # 抉择
> 如今开源的代码已经成为了软件知识传递过程中的一种最佳工具和载体。开源的内在价值取决于开源的代码在时间，空间和不同人群间的无损穿流力。我们也由此相信，任何一个开源项目的创建者在明白了开源的生命力和创新力来源于代码的穿流力之后，他们应该可以在他们所处的环境里去做出一个理智的选择，除非他们已经放弃了开源的初衷。

Reviewer: ```suggestion
## 抉择

如今开源的代码已经成为了软件知识传递过程中的一种最佳工具和载体。开源的内在价值取决于开源的代码在时间，空间和不同人群间的无损穿流力。我们也由此相信，任何一个开源项目的创建者在明白了开源的生命力和创新力来源于代码的穿流力之后，他们应该可以在他们所处的环境里去做出一个理智的选择，除非他们已经放弃了开源的初衷。
```

> 
> 

Reviewer: ```suggestion
---
title:  代码穿流-迷失的开源乌托邦之三
date: 2022-02-25
author: 狄安
categories: 狄安
tags:
    - 开源
    - 开源协同
---
```

玩开源的程序员，大多了解 Linus 曾经说过的那句霸气无比的话，“Talk is cheap. Show me the code”。

<!-- more -->

## 代码胜于雄辩

这是 Linus 于 2000 年 8 月 25 日在社区邮件列表上讨论关于 Linux 内核线程优化问题时，其中一个人提出了个他认为非常高效的方案，而 Linus 认为这个方案不够好，于是他在回信中就写下了这句在今天足以堪称经典的话，这也令很多程序员感觉非常惬意。

从此，每当程序员们产生各种言语纷争而又一时无果时，往往便以这句话来结束唾沫横飞的争论。一方面，我们看到的是程序员文化中“与其你口头zhuangbility，不如代码 Niuabiity”的内涵；另一方面，我们也可这样理解：代码胜过一切含糊不清的争论。因为平常的语言沟通，即使一个明确的想法，即使用同样的字词，却因为语境或者语法造句等因素，让自己原本想表达的构想无法被对方精准理解和捕捉，从而产生各种歧义和误解。而写出来的程序因为代码规则的标准性和编译后的可运行性产生了所见即所得的效果，让代码就不言自证了其原先想法的清晰性，正确性或者优越性。当一切用可见的代码及运行结果来说话，争论自然就偃旗息鼓。

> Facebook 创始人扎克伯格也曾在自己公司上市的前一天发表的上市演讲中说过了类似的话，他当是这样说的：“Code wins argument”，即“代码胜于雄辩。” 其言下之意是想说两层意思：1.代码是一种比平常语言更能精确表达思想的工具。2.代码运行出来的结果可以见证想法的恰当与否。

Reviewer: ```suggestion
Facebook 创始人扎克伯格也曾在自己公司上市的前一天发表的上市演讲中说过了类似的话，他当是这样说的：“Code wins argument”，即“代码胜于雄辩。” 其言下之意是想说两层意思：

1. 代码是一种比平常语言更能精确表达思想的工具。
2. 代码运行出来的结果可以见证想法的恰当与否。
```

1. 知识的显性化
2. 知识的去粘性

进而实现知识的无损耗传递。程序代码则作为人类知识领域的一部分，当然也存在同样的显性化和去粘性问题。

人类的显性知识是可以通过文字、数据、代码、图像等方式进行传递和学习的；而隐形知识则属于人们在长期的学习工作实践中获得并积累下来，与个体的经历及体验有关，或者只能从结果推测，且大多只能意会却难以用语言明示或学习的知识。
对于显性知识下的隐形知识，在翻译美国作家菲茨•吉拉德的名作《了不起的盖茨比》时，作为译者的日本著名作家春上村树在对于如何翻译好开头和结尾、该如何表达原作者想在语言中表达的情绪时，曾经如此纠结过：

> 读上几遍还是只能感慨文字的美，每一个词都有丰富的含义和实质，既有深刻的暗示性，犹如乙醚般清淡，想要捕捉时它却从指尖溜走。 从这段评论中，我们可以看出在日常语言下，即使对于一个专业的作家而言，知识显性化的提取也是非常高难度。当然，菲茨•吉拉德用显性语言来故意蕴藏他想表达的隐含之美是另外一层艺术创作境界，而对于可以我们得出的一个结论是：日常语言对于知识的显性化传递是无法做到足够清晰的，同时，那些受高度语境限制且不确定的语言则更会成为粘性知识。所谓粘性知识是指必须通过近距离的观察和沟通甚至面对面的jiaotan通才能了解获得的知识。

 大家应该知道，有个一个古老的游戏叫“拷贝不走样”。百度百科中是这样描述的：规则是先给第一个人看他要表演的话语或素材，第二个表演给第三个看，依次下传。最后一人要把他理解的意思讲出来，猜中则胜利。该游戏起源于曾经在上海电视台热播的综艺节目《智力大冲浪》。节目最早的版本其实是，人们隔着厚厚的移动墙向下一个选手传与该词相关的动作，然后最后一个人说出第一个人做的姿势的含义，往往笑倒一片观众，拷贝不走样的游戏性在于“拷贝走样”。在活动中，观众可以亲眼目睹动作渐渐走样，这是因为，清楚知道词语的 B 不一定能表达相同信息给后面的 C，动作的模仿中关键的“细节”容易遗漏，或是被 C、D、E 等人再创造而传承。

实际上，这个游戏的隐含意义就是粘性知识在信息的传递过程中，由于传递者自身及语言，动作姿势的模糊性及多义性，信息会被迅速衰减。从这个游戏的结果，我们大致可以得出一个粗浅的结论：粘性知识的传播是距离的衰减函数。

编译成二进制且不提供源代码的程序，虽然通过运行可以看到显性结果或者使用，但事实上因为代码的不可见就是一种典型的隐性且粘性的知识。传统的软件提供商在提供可运行的软件时会附带软件使用手册或者技术文档来局部显性化，也会通过一些面对面的交流沟通（这些往往以培训，现场指导等形式出现）来局部去粘性。但最直接和最透彻的显性化无疑就是提供源代码。当初自由软件运动和随后兴起的开源运动的初心，就是为了让代码知识显性化的无损穿流。

>   
> 

Reviewer: ```suggestion
  
```

> 所以开源软件实践中，程序员通过代码编写将自己的想法精确表达出来，并且通过代码开源行动开放共享出来，就是一次非常典型的知识显性化过程，随后通过社区开展的各类活动如论坛,meetup,会议等线上线下活动进行知识的去粘性化，进而让显性化的代码知识在开源社区的不同成员间无损穿流达到分享传递和协同，让不同程序员的匠性得以聚合，最终达到社会化协同创新。

Reviewer: ```suggestion
所以开源软件实践中，程序员通过代码编写将自己的想法精确表达出来，并且通过代码开源行动开放共享出来，就是一次非常典型的知识显性化过程，随后通过社区开展的各类活动如论坛，meetup，会议等线上线下活动进行知识的去粘性化，进而让显性化的代码知识在开源社区的不同成员间无损穿流达到分享传递和协同，让不同程序员的匠性得以聚合，最终达到社会化协同创新。
```
